clash-prelude/src/Clash/HaskellPrelude.hs

```diff
@@ -14,11 +15,27 @@ splitAt, tail, take, unzip, unzip3, zip, zip3, zipWith, zipWith3.
 {-# OPTIONS_HADDOCK show-extensions, not-home #-}
 
 module Clash.HaskellPrelude
-  (module Prelude)
+  (module Prelude, (&&), (||), not)
 where
 
 import Prelude hiding
   ((++), (!!), concat, concatMap, drop, even, foldl, foldl1, foldr, foldr1, head, init,
    iterate, last, length, map, odd, repeat, replicate, reverse, scanl, scanr, splitAt,
    tail, take, unzip, unzip3, zip, zip3, zipWith, zipWith3, undefined, (^),
-   getChar, putChar, getLine)
+   getChar, putChar, getLine, (&&), (||), not)
+
+import qualified Prelude
+import GHC.Magic (noinline)
+
+infixr 3 &&
+
+(&&) :: Bool -> Bool -> Bool
+(&&) = noinline (Prelude.&&)
+
+infixr 2 ||
+
+(||) :: Bool -> Bool -> Bool
+(||) = noinline (Prelude.||)
+
+not :: Bool -> Bool
+not = noinline Prelude.not
```

Comment:
Perhaps a comment on why we put the `noinline` there?

Reply:
That's a helpful note. I'll add one of my own in the source

Reply:
@martijnbastiaan It's done, but it appears above this span of code so the comment doesn't appear as outdated